service/device/registry/internal/src/main/java/org/eclipse/kapua/service/device/registry/connection/option/internal/DeviceConnectionOptionServiceImpl.java

```diff
@@ -68,11 +70,9 @@ public DeviceConnectionOption update(DeviceConnectionOption deviceConnectionOpti
         AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
         PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
         authorizationService.checkPermission(permissionFactory.newPermission(DeviceDomains.DEVICE_CONNECTION_DOMAIN, Actions.write, deviceConnectionOptions.getScopeId()));
-        DeviceConnectionOptionListResult list = query(query);
-        for (DeviceConnectionOption deviceConnectionOptionDB : list.getItems()) {
-            if (deviceConnectionOptionDB.getReservedUserId() != null && deviceConnectionOptionDB.getReservedUserId().equals(deviceConnectionOptions.getReservedUserId())) {
-                throw new UserAlreadyReservedException(deviceConnectionOptions.getScopeId(), deviceConnectionOptions.getId(), deviceConnectionOptions.getReservedUserId());
-            }
+        query.setPredicate(new AttributePredicateImpl<>(DeviceConnectionAttributes.RESERVED_USER_ID, deviceConnectionOptions.getReservedUserId()));
```

Comment:
The logic should be inverted. If you look for an entry with the same Reserved User ID, you would also find the reservation you're trying to change: basically this check will prevent any update. On the other hand, you should look for any reservation for the same device AND a different user name. If such one exists, you can't add a new one for the same user.